### PR TITLE
Improve docs navigation

### PR DESCRIPTION
**Pull request by benolayinka**
#### Summary

Closes https://github.com/TheThingsIndustries/lorawan-stack-docs/issues/213

New docs nav, with responsive sidebar. Looks like this below 1200px:

Bildschirmfoto 2021-03-03 um 17 12 12

Above 1200px a TOC appears:

Bildschirmfoto 2021-03-03 um 17 12 19

TODO:
- [x] We may have to remove tabs in some places, because any headings within tabs shortcodes won't appear in the right hand navigation. Since we have right hand navigation, we can use headings and allow users to skip to the correct section (have a console heading and a cli heading).
- [x] Make maximum depth 2 (fix cloud integrations) - let's go through and condense instructions which need to be followed in order in to one page, the cloud integrations are a perfect example. I've done it for Akenza as an example. @nejraselimovic 
- [x] Add text in overviews, since these pages are now visible (no more cards).
- [x] Can't have shortcodes in headings - fixed by custom ToC
- [x] Clean up css (ill do this)
- [x] Make headings more descriptive

#### Screenshots

Bildschirmfoto 2021-02-18 um 14 31 54
Bildschirmfoto 2021-02-18 um 14 31 37

### Notable architectural changes

- Three column view, left side nav always visible, right side table of contents jumps to heading anchors
- Removed the themes folder, everything is now in root. It was always confusing what belonged where.
- Reorganize css and partials in smaller files where possible
- Use Hugo's modules to mount `node_modules` in `css/vendor` and `js/vendor`

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

**Comment by benolayinka** [comment]
@nejraselimovic i tagged you in a few TODOs i could use your help on 👏🏽

**Review by johanstokking — COMMENTED** [review]
Looks great. Please request review when ready.

I'm okay with having rather more than less items in the sidebar. I would propose making it more condensed and smaller font size. The biggest problem I have today that I cannot find things in the docs, and so a sidebar with a ToC is very helpful.

**Comment by johanstokking** [comment]
Small issue with the header; the banner isn't the full width:

Screenshot 2021-02-18 at 15 54 14

**Comment by benolayinka** [comment]
> * The banner needs some tweaking. Let's make sure the banner keeps working because we probably need it in the future for something
> * I find the differences in the font size a bit too much; the text in the sidebar is quite small, compared to the body and headings
> * Rename title **Cloud** to **The Things Stack Cloud** (http://localhost:1313/getting-started/cloud-hosted/)
> 
> I didn't review the code (yet), let me know if that is needed

Good catch - fixed the banner, bumped up the font size (to 0.95 of main) and renamed Cloud to TTS Cloud.

**Comment by pierrephz** [comment]
@benolayinka this is a very nice improvement

I have now two things to say:

- let's use a third blue (#E5ECFF) to highlight the current option selected on the nav (see screenshot). Having only a strong blue and white is becoming a bit too agressif.

Screenshot 2021-02-25 at 10 39 36

- from Johan's screenshot, it seems like the nav is not fixed. It should stay fixed on the left while you scroll, like that it's always usable even if you are at the bottom of the page. You don't have to scroll up to find the nav back

**Review by pierrephz — CHANGES_REQUESTED** [review]
See my comment above

File: doc/assets/css/sidebar/sidebar.scss [hunk 1]
```diff
@@ -0,0 +1,50 @@
+.sidebar-left {
+    order: 0;
+
+    @media (min-width: $desktop) {
+    border-right: 1px solid #e9ecef;
+    position: -webkit-sticky;
+    position: sticky;
+    top: 2 * $content-padding-top;
+    align-self: flex-start; /* Fix for sticky in flex */
```
Comment (kschiffer) [inline doc/assets/css/sidebar/sidebar.scss:9]:
```suggestion
    align-self: flex-start; // Fix for sticky in flex
```

Nit

File: doc/assets/css/theme.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
+@charset "utf-8";
+
+// Node_modules mounted to css/vendor and js/vendor in config.toml
```
Comment (kschiffer) [inline doc/assets/css/theme.scss:3]:
```suggestion
// `node_modules` mounted to css/vendor and js/vendor in config.toml
```

Nit

File: doc/layouts/_default/list.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+</h1>
+
+<div class="content">
+	{{ .Content }}
+</div>
+{{ if and .Pages (not (eq .Parent .FirstSection)) }}
+{{ partial "pages-cards.html" .Pages }}
+{{ end }}
+{{ partial "prev-next-in-section" . }}
+{{ partial "feedback" . }}
```
Comment (kschiffer) [inline doc/layouts/_default/list.html:27]: Please decide on one indentation style, it's currently not consistent. I'd suggest 2 spaces.

**Review by kschiffer — COMMENTED** [review]
I'm currently unable to run the dev server due to:
```
go run -tags extended github.com/gohugoio/hugo server -s doc --environment gh-pages
../../go/pkg/mod/github.com/gohugoio/hugo@v0.81.0/tpl/internal/go_templates/texttemplate/helper.go:11:2: package io/fs is not in GOROOT (/usr/local/Cellar/go/1.15.6/libexec/src/io/fs)
make: *** [server] Error 1
```
So I cannot finish the review.

**Review by htdvisser — COMMENTED** [review]
Maybe add CODEOWNERS for `doc/assets/css`, otherwise those files don't get reviewed.

**Comment by benolayinka** [comment]
> See my comment above

The left scrollbar is fixed, it's just that when the banner is showing it messes up the vertical padding on the body that is built in to bulma. This is hard to fix, because it's done with variables in bulma's css, and the banner is toggled in javascript. I'm going to leave this bug in to live with for now. Dismissing the banner fixes it.

**Review by nejraselimovic — APPROVED** [review]

**Comment by benolayinka** [comment]
@johanstokking i went back to small font for the left and right, mimicking the styling here -> https://doks.netlify.app/docs/prologue/introduction/

It looks much cleaner as the left side menu expands, preventing the text from wrapping. Give it one last look and see what you think.

@kschiffer do you still have the error preventing you from running the docs?

**Comment by johanstokking** [comment]
> It looks much cleaner as the left side menu expands, preventing the text from wrapping. Give it one last look and see what you think.

I find it too small. It's so small that it becomes annoying to read and to navigate.

Also comparing it to your reference, it's much smaller:

Screenshot 2021-03-03 at 11 45 02

File: doc/assets/css/common/variables.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+$blue-secondary: #0039CE;
+$blue-alt: #D9E4FF;
+$white: #FFFFFF;
+$graphite: #292929;
+$white-bis: #FBFBFB;
+$white-ter : #ECECEC;
+$red: #F9595A;
+$blue-bg: #E5ECFF;
+$gray: #AEAEAE;
+$red: #FF5C5C;
```
[the remark this answers is not in the document]
Comment (pierrephz) [reply doc/assets/css/common/variables.scss:13]: - for body text: use `$graphite` (rn we are using `$gray`)
- $blue-secondary is: #3D73FF

**Comment by benolayinka** [comment]
> > It looks much cleaner as the left side menu expands, preventing the text from wrapping. Give it one last look and see what you think.
> 
> I find it too small. It's so small that it becomes annoying to read and to navigate.
> 
> We have enough horizontal space to make the menu on the left wider.
> 
> Also comparing it to your reference, it's much smaller:
> 
> Screenshot 2021-03-03 at 11 45 02

Okay. I changed it back, and made the left sidebar slightly larger on screens up to bulma's widescreen query (1200px):

Bildschirmfoto 2021-03-03 um 17 12 12

Above 1200px there's enough space available that it looks alright:

Bildschirmfoto 2021-03-03 um 17 12 19

**Review by johanstokking — CHANGES_REQUESTED** [review]
Font size is OK for me now

I didn't look at the code because frontend tech is not my cup of tea.

I do see many content changes; it seems like this is not properly rebased. Holding merge until that is fixed.

**Review by pierrephz — APPROVED** [review]

**Comment by johanstokking** [comment]
I still don't really get the diff here. Why https://github.com/TheThingsIndustries/lorawan-stack-docs/pull/223/files#diff-44ee0dd4f3e20fbaa46ca55af60ab936cf350b868944abbe0882f8079c5d77b2? Is that part of this PR really?

**Comment by benolayinka** [comment]
> I still don't really get the diff here. Why https://github.com/TheThingsIndustries/lorawan-stack-docs/pull/223/files#diff-44ee0dd4f3e20fbaa46ca55af60ab936cf350b868944abbe0882f8079c5d77b2? Is that part of this PR really?

Editorconfig changes are re this review comment - https://github.com/TheThingsIndustries/lorawan-stack-docs/pull/223#discussion_r582703568

Codeowner change is re https://github.com/TheThingsIndustries/lorawan-stack-docs/pull/223#pullrequestreview-598549976

**Comment by benolayinka** [comment]
> I still don't really get the diff here. Why https://github.com/TheThingsIndustries/lorawan-stack-docs/pull/223/files#diff-44ee0dd4f3e20fbaa46ca55af60ab936cf350b868944abbe0882f8079c5d77b2? Is that part of this PR really?

Ah sorry, the anchor link didn't show for me at first. The changes to the cloud integrations are re: TODO :

`Make maximum depth 2 (fix cloud integrations) - let's go through and condense instructions which need to be followed in order in to one page, the cloud integrations are a perfect example. I've done it for Akenza as an example. @nejraselimovic`

Changing arbitrary depth sections in to maximum depth 2, taking advantage of the right side TOC instead of splitting pages in to separate subsections.

**Review by johanstokking — APPROVED** [review]
OK great stuff.

I'm assuming that you checked that links are changed because of the files being moved, that `aliases` are set

**Review by adriansmares — APPROVED** [review]

**Review by neoaggelos — APPROVED** [review]

**Review by htdvisser — APPROVED** [review]

**Pull request merged**